### PR TITLE
Fixed: `no-missing-end-of-source-newline` warning positioning when using SugarSS parser.

### PR DESCRIPTION
**Pull request by evilebottnawi**
- [ ] Adding a rule: https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#write-the-rule
- [ ] Adding an option: https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-options-to-existing-rules
- [x] Fixing a bug: https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-bugs

Ref https://github.com/stylelint/stylelint/issues/1892

### CODE REVIEW

File: src/rules/no-missing-end-of-source-newline/index.js [hunk 1]
```diff
@@ -15,16 +15,15 @@ export default function (actual) {
...
-      report({
-        message: messages.rejected,
-        node: root,
-        index: root.toString().length - 1,
-        result,
-        ruleName,
-      })
-    }
+    const sourceCss = root.source.input.css
+    if (sourceCss === "" || sourceCss.slice(-1) === "\n") { return }
```
Comment (davidtheclark) [inline src/rules/no-missing-end-of-source-newline/index.js]: I mean, what code fixes the bug?

Comment (evilebottnawi) [inline src/rules/no-missing-end-of-source-newline/index.js]: @davidtheclark https://github.com/stylelint/stylelint/pull/1958 fix bug, i only add tests and speedup rule

Comment (evilebottnawi) [inline src/rules/no-missing-end-of-source-newline/index.js]: sorry for ambiguity

**Pull request merged**